### ISSUE TITLE
Conflict: BetterTouchTool keyboard shortcuts trigger Mouseless overlay

### ISSUE THREAD

**Issue by archetyped** [croian/mouseless-issues#106]
### mouseless version

0.2.2

### macOS version

15.1.1

### Other system input apps

BetterTouchTool

### What happened?

When keyboard shortcuts containing the CMD key are executed by [BetterTouchTool](https://www.folivora.ai/) (BTT), Mouseless' overlay is triggered.

Only keyboard shortcuts containing just the CMD key (no other modifier keys) such as `CMD+W`, `CMD+R`, etc. appear to trigger Mouseless' overlay.

### Steps to Reproduce

In BTT:

1. Create trigger.
2. Add `Send Keyboard Shorcut` action to trigger.
3. Set keyboard shortcut to perform (e.g. `CMD+W`).
4. Run action → Mouseless overlay is displayed.

**Comment by croian** [comment]
Thank you for the report @archetyped, I'll see what I can do about this, in tandem with #101.

**Comment by croian** [comment]
Hi @archetyped, I haven't fully investigated / attempted to repro this yet, but I did change the event tap precedence in the current [preview release](https://github.com/croian/mouseless-issues/releases/tag/v0.3.0-preview.0).

If you have the chance to test out the new build, I'm eager to know if the issue seems fixed! I did learn that some conflicts with Karabiner will not be solvable without a lower-level rewrite of the HID-related code, but hopefully that's not the case here.

Note that the order you start up the apps may affect precedence.

**Comment by archetyped** [comment]
Hi @croian, unfortunately the trial has ended in the interim and the preview release will not run.

Any other options to test the preview release?

Feedback: A 7 day trial is quite short.  A longer demo period (e.g. 30 days) would allow provide time for an app to well-integrated into users' workflows, incentivizing purchase when the demo period has ended.

**Comment by croian** [comment]
@archetyped I agree about the trial period duration, and I appreciate the feedback.  I'd like to make it longer very soon, as well as renew all the trials that have expired during these early days -- just need to get the logistics in place to do so.

in the meantime -- if you send me your trial info found in `/Users/YOUR_USERNAME/Library/Containers/net.sonuscape.mouseless/Data/.mouseless/licenses/trial_license`, i can renew your trial period.

**Comment by archetyped** [comment]
> @archetyped I agree about the trial period duration, and I appreciate the feedback. I'd like to make it longer very soon, as well as renew all the trials that have expired during these early days -- just need to get the logistics in place to do so.
> 
> in the meantime -- if you send me (via email at https://mouseless.click/support) your trial info found in `/Users/YOUR_USERNAME/Library/Containers/net.sonuscape.mouseless/Data/.mouseless/licenses/trial_license`, i can renew your trial period.

I appreciate the workaround, but I prefer to avoid associating additional personal data (e.g. email address, etc.) with the trial license data (machine ID, etc.), so I will wait until you are able to extend/renew trials globally, and will gladly test the updated version then to see if this issue has been fixed.

Thanks.

**Comment by croian** [comment]
@archetyped no problem.  I'll tag you here after that's done.

**Comment by archetyped** [comment]
I installed Mouseless v0.3.0-preview.1 on a different machine and was able to do more testing.

### Keyboard shortcut events sent by external applications

* Example shortcut: `cmd+t`
* Mouseless settings:
    * `show overlay`: `CommandLeft tap`
    * `Tap threshold (ms)`: `200`
---
* BetterTouchTool: ML overlay triggered.
    * `Tap threshold (ms)`: Overlay triggered until value lowered to 2ms or lower.
* Hammerspoon: ML overlay not triggered.
* Karabiner Elements: ML overlay not triggered.

#### Notes

* An option to trigger Mouseless' overlay via a key multi-tap (e.g. `CommandLeft tap-2`, `CommandLeft tap-3`, etc.) would resolve the vast majority of keyboard event conflicts and still trigger the overlay quickly.
* Current workaround: `show overlay` is set to `cmd+alt+ControlLeft tap`, which is mapped to the Caps Lock key via Karabiner (key combo is currently less-used on system, so fewer chances for inadvertent triggering).
* I'll try to stay aware of updated releases to test while this 7 day trial is active.

**Comment by croian** [comment]
thank you for the extra testing and info, @archetyped!  i should have modifier double and triple taps in very soon.  based on what you've said it seems like this conflict is probably similar in nature to the one with Karabiner, which will require a lower-level rewrite of relevant code to fully alleviate

**Comment by croian** [comment]
@archetyped another user may have found [a workaround](https://github.com/croian/mouseless/discussions/183#discussioncomment-12590207) for this issue that lets you keep your desired hotkey in BTT
